### PR TITLE
upload recordings now disables flight mode and uploads

### PR DESCRIPTION
**Pull request by gferraro**
recordings cannot be uploaded while a recording is being made
deleting recordings will cancel a current recording and delete it

### CODE REVIEW

**Comment by timhot** [comment]
Can I just check what you are doing here.  Are you now allowing or preventing a uploading during a recording? The reason I ask (but you may already know this) is that uploading (at least over mobile data) causes strong interference noise on any concurrent recording.

**Comment by gferraro** [comment]
> Can I just check what you are doing here. Are you now allowing or preventing a uploading during a recording? The reason I ask (but you may already know this) is that uploading (at least over mobile data) causes strong interference noise on any concurrent recording.

I am preventing uploading of recordings while recordings are being made.

**Comment by timhot** [comment]
great - thanks

File: app/src/main/java/nz/org/cacophony/birdmonitor/views/ManageRecordingsFragment.java [hunk 1]
```diff
@@ -118,9 +118,13 @@ public void displayOrHideGUIObjects() {
...
-
-
-        if (!Util.isNetworkConnected(getActivity().getApplicationContext())) {
+        if (RecordAndUpload.isRecording) {
+            Toast.makeText(this.getContext(), R.string.upload_currently_recording, Toast.LENGTH_SHORT).show();
+            return;
+        }
+        if (Util.isAirplaneModeOn(getContext())) {
+            Util.disableFlightMode(getContext());
+        } else if (!Util.isNetworkConnected(getActivity().getApplicationContext())) {
```
Comment (CameronRP) [inline app/src/main/java/nz/org/cacophony/birdmonitor/views/ManageRecordingsFragment.java:127]: Don't we want this as an `if` not an `else if`? 
If Airplane mode was on you still want to check that you have an network connection after Airplane mode turns off?

Comment (gferraro) [reply app/src/main/java/nz/org/cacophony/birdmonitor/views/ManageRecordingsFragment.java:127]: It actually waits for an internet connection in  Util.uploadFilesUsingUploadButton
so this check is just to disable flight mode, but if it wasn't enabled to see if we have an active connection.

**Pull request merged**